x-pack/legacy/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/create_analytics_form/create_analytics_form.tsx

```diff
@@ -116,16 +125,50 @@ export const CreateAnalyticsForm: FC<CreateAnalyticsFormProps> = ({ actions, sta
     }
   };
 
-  // Regression supports numeric fields. Classification supports numeric, boolean, text, keyword and ip.
-  const shouldAddFieldOption = (field: Field) => {
+  // Regression and Classification support numeric, boolean, text, keyword, and ip types
+  // Outlier detection supports numeric or boolean types
+  const shouldAddAsAnalyzedFieldsOption = (field: Field) => {
     if (field.id === EVENT_RATE_FIELD_ID) return false;
 
     const isNumerical = NUMERICAL_FIELD_TYPES.has(field.type);
-    const isSupportedByClassification =
-      isNumerical || SUPPORTED_CLASSIFICATION_FIELD_TYPES.has(field.type);
+
+    if (jobType === JOB_TYPES.OUTLIER_DETECTION)
+      return isNumerical || field.type === ES_FIELD_TYPES.BOOLEAN;
+    if (jobType === JOB_TYPES.REGRESSION || jobType === JOB_TYPES.CLASSIFICATION)
+      return isNumerical || SUPPORTED_ANALYSIS_FIELD_TYPES.has(field.type);
+  };
+
+  // Regression supports numeric fields. Classification supports numeric and boolean.
+  const shouldAddAsDepVarOption = (field: Field) => {
+    if (field.id === EVENT_RATE_FIELD_ID) return false;
+
+    const isNumerical = NUMERICAL_FIELD_TYPES.has(field.type);
+    const isSupportedByClassification = isNumerical || field.type === ES_FIELD_TYPES.BOOLEAN;
 
     if (jobType === JOB_TYPES.REGRESSION) return isNumerical;
-    if (jobType === JOB_TYPES.CLASSIFICATION) return isNumerical || isSupportedByClassification;
+    if (jobType === JOB_TYPES.CLASSIFICATION) return isSupportedByClassification;
+  };
+
+  const onCreateOption = (searchValue: string, flattenedOptions: Option[]) => {
+    const normalizedSearchValue = searchValue.trim().toLowerCase();
+
+    if (!normalizedSearchValue) {
+      return;
+    }
+
+    const newOption = {
+      label: searchValue,
+    };
+
+    // Create the option if it doesn't exist.
+    if (
+      flattenedOptions.findIndex(
```

Comment:
Nit: Could use `some()` instead.

Reply:
Updated in 5fb02c50cabc18a94c4952adee9d279bea7b54ac